pkg/resources/common/common.go

```diff
@@ -51,40 +51,30 @@ func IsModifyingLabel(oldLabels, newLabels map[string]string, label string) bool
 	return false
 }
 
-// cachedVerbChecker is used for caching if a request for a non-namespaced gvr with specified name has the given overrideVerb.
+// CachedVerbChecker is used for caching if a request for a non-namespaced gvr with specified name has the given overrideVerb.
 // This is meant to eliminate the need to perform multiple calls to the provided SubjectAccessReview for the overrideVerb.
-type cachedVerbChecker struct {
-	request            *admission.Request
-	name               string
-	sar                authorizationv1.SubjectAccessReviewInterface
-	gvr                schema.GroupVersionResource
-	overrideVerb       string
+// Each CachedVerbChecker is unique to the initial set up. If the caller needs to change what it is checking
+// (different verb, resource name, resource type) a new CachedVerbChecker must be created.
+// A CachedVerbChecker should not be shared between admitters. Each admitter must request a new CachedVerbChecker.
+type CachedVerbChecker struct {
+	Request            *admission.Request
+	Name               string
+	SAR                authorizationv1.SubjectAccessReviewInterface
+	GVR                schema.GroupVersionResource
+	OverrideVerb       string
 	hasVerbBeenChecked bool
 	hasVerb            bool
 }
 
-// NewCachedVerbChecker creates a new cachedVerbChecker.
-func NewCachedVerbChecker(req *admission.Request, name string, sar authorizationv1.SubjectAccessReviewInterface, gvr schema.GroupVersionResource, verb string) *cachedVerbChecker {
-	return &cachedVerbChecker{
-		request:            req,
-		name:               name,
-		sar:                sar,
-		gvr:                gvr,
-		overrideVerb:       verb,
-		hasVerbBeenChecked: false,
-		hasVerb:            false,
-	}
-}
-
 // IsRulesAllowed checks if the request has permissions to create the rules provided.
 // If not it checks if the user has the needed overrideVerb to bypass the check.
-func (c *cachedVerbChecker) IsRulesAllowed(rules []v1.PolicyRule, resolver validation.AuthorizationRuleResolver, namespace string) error {
-	err := auth.ConfirmNoEscalation(c.request, rules, namespace, resolver)
+func (c *CachedVerbChecker) IsRulesAllowed(rules []v1.PolicyRule, resolver validation.AuthorizationRuleResolver, namespace string) error {
```

Comment:
Right now this returns an error if the user is trying to escalate but has the verb for override. I think that's a bit of a confusing UX since the presence of the override verb should allow the action - but the function requires a separate check for the consuming area to see that happened. 

In short, right now in override scenarios this returns an error - I think it should not.